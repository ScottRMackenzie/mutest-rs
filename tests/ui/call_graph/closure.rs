//@ print-call-graph
//@ print-targets
//@ stdout
//@ stderr: empty

fn impl_closure() {}

fn make_closure_call<F: Fn()>(f: F) {
    f();
}

#[test]
fn test() {
    let f = || impl_closure();
    f();

    make_closure_call(|| impl_closure());
}
