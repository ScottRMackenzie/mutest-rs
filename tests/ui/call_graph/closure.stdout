
@@@ call graph @@@

entry points:

test test
  -> make_closure_call::<{closure@tests/ui/call_graph/closure.rs:17:23: 17:25}> at tests/ui/call_graph/closure.rs:8:1: 8:36 (#0)
  -> test::{closure#0} at tests/ui/call_graph/closure.rs:14:13: 14:15 (#0)

nested calls at distance 1:

make_closure_call::<{closure@tests/ui/call_graph/closure.rs:17:23: 17:25}> at tests/ui/call_graph/closure.rs:8:1: 8:36 (#0)
  -> test::{closure#1} at tests/ui/call_graph/closure.rs:17:23: 17:25 (#0)
test::{closure#0} at tests/ui/call_graph/closure.rs:14:13: 14:15 (#0)
  -> impl_closure at tests/ui/call_graph/closure.rs:6:1: 6:18 (#0)

nested calls at distance 2:

test::{closure#1} at tests/ui/call_graph/closure.rs:17:23: 17:25 (#0)
  -> impl_closure at tests/ui/call_graph/closure.rs:6:1: 6:18 (#0)

@@@ targets @@@

tests -(1)-> impl_closure at tests/ui/call_graph/closure.rs:6:1: 6:18 (#0)
  (1) test

tests -(0)-> make_closure_call at tests/ui/call_graph/closure.rs:8:1: 8:36 (#0)
  (0) test

targets: 2 total; 2 safe; 0 unsafe (0 tainted)
